test-harness/test-harness-macros/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+/// into a variable will be generated. The name of the variable will match the
+/// one used as argument.
+///
+/// This will "consume" all the tokens that are of type `Package`.
+fn load_package(function: &mut syn::ItemFn) {
+    let found_packages_vars =
+        process_arguments::<Package>(function, 1).unwrap_or_else(|err| panic!("{err}"));
+
+    let Some(package_binding_name) = found_packages_vars.first() else {
+        // If there are no variables with `Package` as its type, then don't load
+        // the `Package`.
+        return;
+    };
+
+    // This env var is set by `cargo miden test`.
+    let package_path = std::env::var("CREATED_PACKAGE").unwrap();
+
+    let load_package: Vec<syn::Stmt> = syn::parse_quote! {
+        let path = #package_path;
+        let bytes = std::fs::read(path).unwrap();
+        let #package_binding_name =
+            <miden_test_harness_lib::reexport::__miden_test_harness_miden_mast_package::Package
+            as miden_test_harness_lib::reexport::__miden_test_harness_Deserialzable>::read_from_bytes(&bytes).unwrap();
+    };
+
+    // We add the required lines to load the generated Package right at the
+    // beginning of the function.
+    for (i, package) in load_package.iter().enumerate() {
+        function.block.as_mut().stmts.insert(i, package.clone());
+    }
+}
+
+fn load_mock_chain(function: &mut syn::ItemFn) {
+    let found_mock_chain =
+        process_arguments::<MockChainBuilder>(function, 1).unwrap_or_else(|err| panic!("{err}"));
+
+    let Some(mock_chain_builder_name) = found_mock_chain.first() else {
+        // If there are no variables with `MockChainBuilder` as its type, then don't load
+        // the `MockChainBuilder`.
+        return;
+    };
+
+    let load_mock_chain_builder: Vec<syn::Stmt> = syn::parse_quote! {
+        let mut #mock_chain_builder_name = miden_test_harness_lib::reexport::miden_testing::MockChainBuilder::new();
+    };
+
+    // We add the required lines to load the generated MockChainBuilder right at the
+    // beginning of the function.
+    for (i, package) in load_mock_chain_builder.iter().enumerate() {
+        function.block.as_mut().stmts.insert(i, package.clone());
+    }
+}
+
+#[proc_macro_attribute]
+pub fn miden_test(
+    _attr: proc_macro::TokenStream,
+    item: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let mut input_fn = parse_macro_input!(item as ItemFn);
+
+    let fn_ident = input_fn.sig.ident.clone();
+    let fn_name = fn_ident.clone().span().source_text().unwrap();
+
+    load_package(&mut input_fn);
+    load_mock_chain(&mut input_fn);
+
+    let function = quote! {
+        miden_test_harness_lib::miden_test_submit!(
+            miden_test_harness_lib::MidenTest {
+                name: #fn_name,
+                test_fn: #fn_ident,
+            }
+        );
+
+        #[cfg(test)]
+        #input_fn
+    };
+
+    TokenStream::from(function)
+}
+
+#[proc_macro_attribute]
+pub fn miden_test_block(
```

Comment:
Let's name this attribute `#[miden_test_suite]`, since that is effectively what it is used to define.